libfreerdp/core/gateway/arm.c

```diff
@@ -56,18 +56,26 @@
 #include <cjson/cJSON.h>
 #endif
 
+#include <time.h>
+#include <string.h>
+
 struct rdp_arm
 {
 	rdpContext* context;
 	rdpTls* tls;
 	HttpContext* http;
+
+	UINT32 gateway_retry;
 };
 
 typedef struct rdp_arm rdpArm;
 
 #define TAG FREERDP_TAG("core.gateway.arm")
 
 #ifdef WITH_AAD
+static const UINT32 max_retries = 60;
```

Comment:
Shouldn't these variable be in settings ?

Reply:
not sure this needs to be configurable. the timeout is currently hard coded to 10 minutes, anything longer will most likely be useless anyway.